src/Core/Components/DataGrid/Columns/SelectColumn.cs

```diff
@@ -0,0 +1,364 @@
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.Web;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// Represents a <see cref="FluentDataGrid{TGridItem}"/> column whose cells render a selected checkbox updated when the user click on a row.
+/// </summary>
+/// <typeparam name="TGridItem">The type of data represented by each row in the grid.</typeparam>
+public class SelectColumn<TGridItem> : ColumnBase<TGridItem>
+{
+    /// <summary>
+    /// List of keys to press, to select/unselect a row.
+    /// </summary>
+    public static string[] KEYBOARD_SELECT_KEYS = ["Enter", "NumpadEnter"];
+
+    private DataGridSelectMode _selectMode = DataGridSelectMode.Single;
+    private readonly List<TGridItem> _selectedItems = new List<TGridItem>();
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="SelectColumn{TGridItem}"/>.
+    /// </summary>
+    public SelectColumn()
+    {
+        Width = "50px";
+        ChildContent = GetDefaultChildContent();
+    }
+
+    /// <summary>
+    /// Gets or sets the content to be rendered for each row in the table.
+    /// </summary>
+    [Parameter]
+    public RenderFragment<TGridItem> ChildContent { get; set; }
+
+    /// <summary>
+    /// Gets or sets the list of selected items.
+    /// </summary>
+    [Parameter]
+    public IEnumerable<TGridItem> SelectedItems
+    {
+        get => _selectedItems;
+        set
+        {
+            if (_selectedItems != value)
+            {
+                _selectedItems.Clear();
+                _selectedItems.AddRange(value);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a callback when list of selected items changed.
+    /// </summary>
+    [Parameter]
+    public EventCallback<IEnumerable<TGridItem>> SelectedItemsChanged { get; set; }
+
+    /// <summary>
+    /// Gets or sets the selection mode (Single or Multiple).
+    /// </summary>
+    [Parameter]
+    public DataGridSelectMode SelectMode
+    {
+        get => _selectMode;
+        set
+        {
+            _selectMode = value;
+
+            if (value == DataGridSelectMode.Single)
+            {
+                KeepOnlyFirstSelectedItemAsync().Wait();
+            }
+
+            RefreshHeaderContent();
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the Icon to be rendered when the row is non selected.
+    /// </summary>
+    [Parameter]
+    public required Icon IconUnchecked { get; set; } = new CoreIcons.Regular.Size20.CheckboxUnchecked().WithColor(Color.FillInverse);
```

Comment:
FluentUI 9 uses a radio for single select row checked/unchecked state. As we have those icons as well, might make sense to use these. And use Checkbox for multi-select. Probably easiest to add two new parameters (SingleSelectIconUnchecked, SingleSelectIconChecked) and rename the existing ones (IconUnchecked -> MultiSelectIconUnchecked, IconChecked -> MultiSelectIconChecked and IconIndeterminate -> MultiSelectIconIndeterminate  )
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/abb41e91-0664-4c0d-aafc-607e3533733d)

Reply:
Done... and GIF above updated.